search_queries_interval.go

```diff
@@ -0,0 +1,43 @@
+// Copyright 2012-present Oliver Eilhard. All rights reserved.
+// Use of this source code is governed by a MIT-license.
+// See http://olivere.mit-license.org/license.txt for details.
+
+package elastic
+
+// IntervalRule represents the generic matching rule interface. Matching Rule is
+// just a Query, but may be used only inside IntervalQuery. An extra method is added
+// just to separate *Rule objects from other queries
+type IntervalRule interface {
```

Comment:
Can we maybe rename that to `IntervalQueryRule`, so that they share a common suffix? Will probably help with intellisense in such a huge package that is `elastic.Interval___`.

Reply:
maybe `IntervalRuleQuery` then?

Reply:
I'd go for `IntervalQueryRule` so that everything "interval query"-related has the same prefix: `IntervalQuery`.